cmd/minishift/cmd/openshift/service.go

```diff
@@ -50,25 +53,78 @@ var serviceCmd = &cobra.Command{
 			atexit.ExitWithMessage(1, "You must specify the name of the service.")
 		}
 
+		host, err := api.Load(constants.MachineName)
+		if err != nil {
+			atexit.ExitWithMessage(1, err.Error())
+		}
+
+		util.ExitIfNotRunning(host.Driver, constants.MachineName)
+
+		ip, err := host.Driver.GetIP()
+		if err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error getting IP: %s", err.Error()))
+		}
+
 		service := args[0]
 
-		url, err := openshift.GetServiceSpec(service, namespace, https)
+		serviceSpec, err := openshift.GetServiceSpec(service, namespace, https)
 		if err != nil {
 			atexit.ExitWithMessage(1, err.Error())
 		}
 
-		if urlMode {
-			fmt.Fprintln(os.Stdout, url)
+		if url {
+			stdOutURL(serviceSpec)
+		} else if inbrowser {
+			openInBrowser(serviceSpec)
 		} else {
-			fmt.Fprintln(os.Stdout, "Opening the service "+namespace+"/"+service+" in the default browser...")
-			browser.OpenURL(url)
+			var data [][]string
+			var urls, weights string
+
+			nodePortURL := serviceSpec.NodePort
+			if nodePortURL != "" {
+				nodePortURL = fmt.Sprintf("%s:%s", ip, nodePortURL)
+			}
+			if serviceSpec.URL != nil {
+				urls = strings.Join(serviceSpec.URL, "\n")
+			}
+			if serviceSpec.Weight != nil {
+				weights = strings.Join(serviceSpec.Weight, "\n")
+			}
+
+			data = append(data, []string{serviceSpec.Namespace, serviceSpec.Name, nodePortURL, urls, weights})
+			table := tablewriter.NewWriter(os.Stdout)
+			table.SetHeader([]string{"Namsepace", "Name", "NodePort", "Route-URL", "Weight"})
+			table.SetBorders(tablewriter.Border{Left: true, Top: true, Right: true, Bottom: true})
+			table.SetCenterSeparator("|")
+			table.AppendBulk(data)
+			table.Render()
 		}
 	},
 }
 
 func init() {
 	serviceCmd.Flags().StringVarP(&namespace, "namespace", "n", "default", "The namespace of the service.")
-	serviceCmd.Flags().BoolVar(&urlMode, "url", false, "Access the service in the command-line console instead of in the default browser.")
+	serviceCmd.Flags().BoolVar(&inbrowser, "inbrowser", false, "Access the service in the default browser.")
+	serviceCmd.Flags().BoolVar(&url, "url", false, "Access the service url in the command-line console.")
 	serviceCmd.Flags().BoolVar(&https, "https", false, "Access the service with HTTPS instead of HTTP.")
 	OpenShiftCmd.AddCommand(serviceCmd)
 }
+
+func openInBrowser(serviceSpec openshift.ServiceSpec) {
+	if serviceSpec.URL != nil {
```

Comment:
I am wondering whether we should allow opening a NodePort in browser. See https://github.com/minishift/minishift/issues/1106#issuecomment-314100093